tests/unit/sagemaker/workflow/test_repack_model_script.py

```diff
@@ -0,0 +1,174 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from sagemaker.workflow import _repack_model
+
+from pathlib import Path
+import shutil
+import tarfile
+import os
+import pytest
+import time
+
+
+@pytest.mark.skip(
+    reason="""This test operates on the root file system
+                            and will likely fail due to permission errors.
+                            Temporarily remove this skip decorator and run
+                            the test after making changes to _repack_model.py"""
+)
+def test_repack_entry_point_only(tmp):
+    model_name = "xg-boost-model"
+    fake_model_path = os.path.join(tmp, model_name)
+
+    # create a fake model
+    open(fake_model_path, "w")
+
+    # create model.tar.gz
+    model_tar_name = "model-%s.tar.gz" % time.time()
+    model_tar_location = os.path.join(tmp, model_tar_name)
+    with tarfile.open(model_tar_location, mode="w:gz") as t:
+        t.add(fake_model_path, arcname=model_name)
+
+    # move model.tar.gz to /opt/ml/input/data/training
+    Path("/opt/ml/input/data/training").mkdir(parents=True, exist_ok=True)
+    shutil.move(model_tar_location, os.path.join("/opt/ml/input/data/training", model_tar_name))
+
+    # create files that will be added to model.tar.gz
+    create_file_tree(
+        "/opt/ml/code",
+        [
+            "inference.py",
+        ],
+    )
+
+    # repack
+    _repack_model.repack(inference_script="inference.py", model_archive=model_tar_name)
```

Comment:
nice, would be nice to also cover `__main__`

Reply:
One of the reasons I moved the logic to a function was because I couldn't figure out how to invoke the `__main__` of a module from within python :) As far as I understand,`__name__` is only set to `__main__` if it's called from a command line. Suggestions welcome